ts/pubsub.js

```diff
@@ -248,17 +235,43 @@ class BasicPubSub extends Pubsub {
    * @param {Peer} peer
    * @param {RPC.Message} msg
    */
-  _processRpcMessage (peer, msg) {
+  async _processRpcMessage (peer, msg) {
     if (this.peerId.toB58String() === msg.from && !this._options.emitSelf) {
       return
     }
+    // Ensure the message is valid before processing it
+    try {
+      const isValid = await this.validate(utils.normalizeOutRpcMessage(msg), peer)
+      if (!isValid) {
+        this.log('Message is invalid, dropping it.')
+        return
+      }
+    } catch (err) {
+      this.log('Error in message validation, dropping it. %O', err)
+      return
+    }
+
+    this._publishFrom(peer, msg)
+  }
 
+  /**
+   * Publish a message sent from a peer
+   * @param {Peer} peer
+   * @param {InMessage} msg
+   * @returns {void}
+   */
+  _publishFrom (peer, msg) {
```

Comment:
Should we validate if we should emit to self here?

Reply:
We weren't doing this before, I believe the current behavior of `emitSelf` is to suppress the events from messages published from self. The naming here is confusing :(

Reply:
you are right. We can also discuss the naming later if you have any suggestion